src/model/paths_types.ml

```diff
@@ -1,128 +1,187 @@
 open Names
 (** {1 Paths} *)
 
+type 'a id = { iv : 'a; ihash : int; ikey : string }
```

Comment:
Why store the hash instead of letting it be recomputed from the key ?

Reply:
just so we only compute it once.